### PR TITLE
Markdown: turn plain URLs into links

### PR DESCRIPTION
**Pull request by danyx23**
By default, plain link like https://domain.com are not shown as links in markdown (see [commonmark spec](https://spec.commonmark.org/0.30/#autolinks)) and instead have to be written in the `[text](https://domain.com)` style. 

This PR adds autolinking to plain links. This is done in two different way
- once for the SimpleMarkdown component that uses react-remark via a remark transformation plugin
- and once for the MarkdownTextWrap component by doing the translation as part of the mapping to IRNodes

In theory it should be possible to use the remark plugin in both cases, but navigating the set of interdependent libraries at the state two years ago is a pain and so I went for the two bespoke versions. When we upgrade our setup to be ESM everywhere we can upgrade to the latest versions of the unified, react-remark et al libraries and try again to use the same code in both cases.

### CODE REVIEW

**Comment by danyx23** [comment]
Current dependencies on/for this PR:
* master
  * **PR #2875** <a href="https://app.graphite.dev/github/pr/owid/owid-grapher/2875" target="_blank">Graphite</a> 
    * **PR #2872** <a href="https://app.graphite.dev/github/pr/owid/owid-grapher/2872" target="_blank">Graphite</a> 
      * **PR #2884** <a href="https://app.graphite.dev/github/pr/owid/owid-grapher/2884" target="_blank">Graphite</a>  👈
        * **PR #2873** <a href="https://app.graphite.dev/github/pr/owid/owid-grapher/2873" target="_blank">Graphite</a> 
          * **PR #2874** <a href="https://app.graphite.dev/github/pr/owid/owid-grapher/2874" target="_blank">Graphite</a> 

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/owid/owid-grapher/2884?utm_source=stack-comment).

**Review by marcelgerber — APPROVED** [review]

File: packages/@ourworldindata/components/src/markdown/remarkPlainLinks.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
+import findAndReplace from "mdast-util-find-and-replace"
+
+export const urlRegex = /https?:\/\/([\w-]+\.)+[\w-]+(\/[\w\- .\+/?:%&=~#]*)?/
```
Comment (marcelgerber) [inline packages/@ourworldindata/components/src/markdown/remarkPlainLinks.ts:3]: Ah, now that I think of it, including `.?:` can be annoying in cases like `Learn more at https://ourworldindata.com/poverty.`.

I know that some other sites do something like "match `.?:` unless they are at the very end of the URL", which seems sensible.

Comment (danyx23) [reply packages/@ourworldindata/components/src/markdown/remarkPlainLinks.ts]: Good catch. I took this opportunity to also make sure we match urls with unicode word characters in the path. 

AFAIK to properly match this with a regex you need to use negative lookbehind assertions and they are not supported in Safari before 16.4 which came out in Spring this year so I instead handle trailing punctuation characters in the replacement function.

**Comment by danyx23** [comment]
### Merge activity

* **Nov 7, 3:55 PM**: @danyx23 started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/owid/owid-grapher/2884).

**Pull request merged**